rllib/utils/test_utils.py

```diff
@@ -372,6 +347,88 @@ def check_compute_single_action(trainer,
                         "({})!".format(action, what, action_space))
 
 
+def check_learning_achieved(tune_results, min_reward, evaluation=False):
+    """Throws an error if `min_reward` is not reached within tune_results.
+
+    Checks the last iteration found in tune_results for its
+    "episode_reward_mean" value and compares it to `min_reward`.
+
+    Args:
+        tune_results: The tune.run returned results object.
+        min_reward (float): The min reward that must be reached.
+
+    Raises:
+        ValueError: If `min_reward` not reached.
+    """
+    # Get maximum reward of all trials
+    # (check if at least one trial achieved some learning)
+    avg_rewards = [(trial.last_result["episode_reward_mean"]
+                    if not evaluation else
+                    trial.last_result["evaluation"]["episode_reward_mean"])
+                   for trial in tune_results.trials]
+    best_avg_reward = max(avg_rewards)
+    if best_avg_reward < min_reward:
+        raise ValueError("`stop-reward` of {} not reached!".format(min_reward))
+    print("ok")
+
+
+def check_train_results(train_results):
+    """Checks proper structure of a Trainer.train() returned dict.
+
+    Args:
+        train_results: The train results dict to check.
+
+    Raises:
+        AssertionError: If `train_results` doesn't have the proper structure or
+            data in it.
+    """
+
+    from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+    from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, \
+        LEARNER_STATS_KEY
+
+    # Assert that all the keys are where we would expect them.
+    for key in ["info", "hist_stats", "timers", "perf", "episode_reward_mean"]:
+        assert key in train_results, \
+            f"'{key}' not found in `train_results` ({train_results})!"
+
+    is_multi_agent = len(train_results["policy_reward_min"]) > 0
```

Comment:
now that everything is consistent, can we check the size of the policy dict or something?
seems like a more direct condition than reward_min.

Reply:
Yeah, you are totally right! We even have a util function that returns `is_multiagent` based on the config. I'll use that here instead.

Reply:
done